tools/oeedger8r/src/Emitter.ml

```diff
@@ -59,6 +61,41 @@ let warn_non_portable_types (fd : func_decl) =
     print_portability_warning_with_recommendation "unsigned long"
       "uint64_t or uint32_t"
 
+let warn_shallow_ptrs (ec : enclave_content) =
+  (* Gather every known parameter. *)
+  let splist =
+    flatten_map (function StructDef s -> s.smlist | _ -> []) ec.comp_defs
+  in
+  let ptrs = List.filter is_ptr splist in
+  List.iter
+    (fun (p, d) ->
+      (* This won't work. Function params can have [out] without
+         [size] or [copy] which leads to [is_marshalled_ptr] being
+         true. Function params have to be tested differently. *)
+      if not (is_marshalled_ptr p) then
```

Comment:
@mingweishih Shouldn't we also consider fields with count attribute as marshalled?